extensions/ftp/src/main/java/org/mule/extension/ftp/internal/sftp/connection/SftpClient.java

```diff
@@ -178,6 +191,42 @@ private void configureHostChecking(Properties hash) throws JSchException {
     }
   }
 
+  private void configureProxy(Session session) {
+    if (proxyConfig != null) {
+      Proxy proxy = null;
+      switch (proxyConfig.getProtocol()) {
+        case HTTP:
+          ProxyHTTP proxyHttp = new ProxyHTTP(proxyConfig.getHost(), proxyConfig.getPort());
+          if (proxyConfig.getUsername() != null) {
+            proxyHttp.setUserPasswd(proxyConfig.getUsername(), proxyConfig.getPassword());
```

Comment:
It would be ignored by the jsch library, but it would be more explicit if both checks are present here.